trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/BookmarkManagerService.java

```diff
@@ -0,0 +1,474 @@
+            @ApiResponse(responseCode = "404", description = "Experiment or bookmark not found", content = @Content(schema = @Schema(implementation = String.class)))
+    })
+    public Response getBookmark(
+            @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+            @Parameter(description = "Bookmark UUID") @PathParam("bookmarkUUID") UUID bookmarkUUID) {
+
+        TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+        if (experiment == null) {
+            return Response.status(Status.NOT_FOUND).entity(NO_SUCH_EXPERIMENT).build();
+        }
+
+        Map<UUID, Bookmark> bookmarks = EXPERIMENT_BOOKMARKS.get(expUUID);
+        if (bookmarks == null || !bookmarks.containsKey(bookmarkUUID)) {
+            return Response.status(Status.NOT_FOUND).entity(EndpointConstants.BOOKMARK_NOT_FOUND).build();
+        }
+
+        return Response.ok(bookmarks.get(bookmarkUUID)).build();
+    }
+
+    /**
+     * Create a new bookmark in an experiment
+     *
+     * @param expUUID
+     *            UUID of the experiment
+     * @param queryParameters
+     *            Parameters for creating the bookmark
+     * @return Response containing the created bookmark
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Create a new bookmark in an experiment", responses = {
+            @ApiResponse(responseCode = "200", description = "Bookmark created successfully", content = @Content(schema = @Schema(implementation = Bookmark.class))),
+            @ApiResponse(responseCode = "400", description = INVALID_PARAMETERS, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = NO_SUCH_EXPERIMENT, content = @Content(schema = @Schema(implementation = String.class)))
+    })
+    public Response createBookmark(
+            @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+            @RequestBody(required = true) QueryParameters queryParameters) {
+
+        if (queryParameters == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_PARAMETERS).build();
+        }
+
+        Map<String, Object> parameters = queryParameters.getParameters();
+        String errorMessage = QueryParametersUtil.validateBookmarkQueryParameters(parameters);
+        if (errorMessage != null) {
+            return Response.status(Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+
+        TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+        if (experiment == null) {
+            return Response.status(Status.NOT_FOUND).entity(NO_SUCH_EXPERIMENT).build();
+        }
+
+        String name = Objects.requireNonNull((String) parameters.get("name")); //$NON-NLS-1$
```

Comment:
This causes a NullPointerException. The server will return an error 500 internal error, but it won't help the FE user to know what the problem. You need to check for null and then response with a proper error (400) and error string.

Reply:
I guess this issue has been already handled here:
```
Map<String, Object> parameters = queryParameters.getParameters();
String errorMessage = QueryParametersUtil.validateBookmarkQueryParameters(parameters);
if (errorMessage != null) {
    return Response.status(Status.BAD_REQUEST).entity(errorMessage).build();
}
```

So if these variables are null, the validator will return the corresponding error code for FE procedures.